server_api.py

```diff
@@ -797,23 +797,23 @@ def edit_image_data(json_obj):
                 return return_msg
             
             #get old image type dir
-            sql = ("SELECT type_dir FROM data_type WHERE type_id=" + str(img_type_id))
-            pure_result = db.query(sql)
+            with DataTypeDao() as dataTypeDao:
+                type_dir = dataTypeDao.getTypeDir(typeId=str(img_type_id))
             try: 
```

Comment:
I think the `try...except` block can be replace by `if type_dir...else`, because the queryOneValue() will return `None` while the SQL result is empty